google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -353,13 +359,19 @@ FirestoreRpc getClient() {
   }
 
   /** Request funnel for all bidirectional streaming requests. */
-  <RequestT, ResponseT> ApiStreamObserver<RequestT> streamRequest(
+  @Override
+  public <RequestT, ResponseT> ApiStreamObserver<RequestT> streamRequest(
       ApiStreamObserver<ResponseT> responseObserverT,
       BidiStreamingCallable<RequestT, ResponseT> callable) {
     Preconditions.checkState(!closed, "Firestore client has already been closed");
     return callable.bidiStreamingCall(responseObserverT);
   }
 
+  @Override
+  public FirestoreImpl getFirestore() {
```

Comment:
I wish this was called differently, but I cannot think of a better name than the already taken `getClient()`. The frequent calls to `firestore.getFirestore()` seems slightly strange to me.

Reply:
Yeah, I'm not crazy about it either.

This method is serving two purposes I can see:
1. Direct access to the Firestore instance for return from a models `getFirestore()` method
2. Access to allow calling `Firestore#batch()` (14 times) or `Firestore#getAll(DocumentReference...)` (1 time)

Regarding 1 above, I don't think there is much we can do there since those `getFirestore()` are part of the public API. For 2, given the necessity in 1 I don't know that it makes sense for us too refactor things to remove the direct dependence on 2 at this time (especially with the upcoming work on BulkWriter).

would it help to rename the usage locations of the `FirestoreRpcContext` from `firestore` to something like `context` or `firestoreRpcContext`? (I didn't do this in my first push, to try and help reduce diff noise.)

Reply:
I think `rpcContext` would be great and is as descriptive as `firestoreRpcContext` in the context of this repo.

Reply:
I'll push a new commit with this change in just a few minutes.

Reply:
New commit pushed